http/param/param.go

```diff
@@ -217,6 +244,23 @@ func (p Parser) parsePathParam(r *http.Request, paramName string, v reflect.Valu
 	return nil
 }
 
+func (p Parser) parseFormParam(r *http.Request, paramName string, v reflect.Value) error {
+	if r.Method != http.MethodPost && r.Method != http.MethodPut && r.Method != http.MethodPatch {
+		return fmt.Errorf("struct's field was tagged for parsing the form parameter (%s) but request method is not POST, PUT or PATCH", paramName)
+	}
+	if err := r.ParseForm(); err != nil {
```

Comment:
testing with `curl`, naively googling how to do form requests with it, `curl -F name=value` sends a different format - `multipart/form-data`. The `ParseForm` handles `application/x-www-form-urlencoded` and `application/octet-stream`.

(my suggestion) Changing this to `ParseMultipartForm` would handle all content types as before + the `multipart/form-data`.

Removing this (and still calling `FormParamFunc` or `PostFormValue` below - which itself calls `ParseMultipartForm` on first call) would also handle all content types as before + the `multipart/form-data`, but would't allow handling errors